client/object_get.go

```diff
@@ -0,0 +1,728 @@
+
+	x.tailPayload = x.tailPayload[read:]
+
+	if len(buf) == read {
+		return read, true
+	}
+
+	// receive next message
+	ok := x.ctxCall.readResponse()
+	if !ok {
+		return read, false
+	}
+
+	// get chunk part message
+	part := x.bodyResp.GetObjectPart()
+
+	var partChunk *v2object.GetObjectPartChunk
+
+	partChunk, ok = part.(*v2object.GetObjectPartChunk)
+	if !ok {
+		x.ctxCall.err = fmt.Errorf("unexpected message instead of chunk part: %T", part)
+		return read, false
+	}
+
+	// read new chunk
+	chunk := partChunk.GetChunk()
+
+	tailOffset := copy(buf[read:], chunk)
+
+	read += tailOffset
+
+	// save the tail
+	x.tailPayload = append(x.tailPayload, chunk[tailOffset:]...)
+
+	return read, true
+}
+
+// ReadChunk reads another chunk of the object payload. Works similar to
+// io.Reader.Read but returns success flag instead of error.
+//
+// Failure reason can be received via Close.
+func (x *ObjectReader) ReadChunk(buf []byte) (int, bool) {
+	return x.readChunk(buf)
+}
+
+func (x *ObjectReader) close(ignoreEOF bool) (*ResObjectGet, error) {
+	defer x.cancelCtxStream()
+
+	if x.ctxCall.err != nil {
+		if !errors.Is(x.ctxCall.err, io.EOF) {
+			return nil, x.ctxCall.err
+		} else if x.remainingPayloadLen > 0 {
+			return nil, io.ErrUnexpectedEOF
+		} else if !ignoreEOF {
+			return nil, io.EOF
+		}
+	}
+
+	return x.ctxCall.statusRes.(*ResObjectGet), nil
+}
+
+// Close ends reading the object and returns the result of the operation
+// along with the final results. Must be called after using the ObjectReader.
+//
+// Exactly one return value is non-nil. By default, server status is returned in res structure.
+// Any client's internal or transport errors are returned as Go built-in error.
+// If Client is tuned to resolve NeoFS API statuses, then NeoFS failures
+// codes are returned as error.
+//
+// Return errors:
+//   *object.SplitInfoError (returned on virtual objects with PrmObjectGet.MakeRaw).
+//
+// Return statuses:
+//   global (see Client docs).
+func (x *ObjectReader) Close() (*ResObjectGet, error) {
+	return x.close(true)
+}
+
+func (x *ObjectReader) Read(p []byte) (int, error) {
+	n, ok := x.readChunk(p)
+	if !ok {
+		res, err := x.close(false)
+		if err != nil {
+			return n, err
+		}
+
+		return n, apistatus.ErrFromStatus(res.Status())
+	}
+
+	if n > x.remainingPayloadLen {
+		return n, errors.New("payload size overflow")
+	}
+
+	x.remainingPayloadLen -= n
+
+	return n, nil
+}
+
+// ObjectGetInit initiates reading an object through a remote server using NeoFS API protocol.
+//
+// The call only opens the transmission channel, explicit fetching is done using the ObjectReader.
+// Exactly one return value is non-nil. Resulting reader must be finally closed.
+//
+// Immediately panics if parameters are set incorrectly (see PrmObjectGet docs).
+// Context is required and must not be nil. It is used for network communication.
+func (c *Client) ObjectGetInit(ctx context.Context, prm PrmObjectGet) (*ObjectReader, error) {
```

Comment:
suggest adding public functionality that allows a reading full object in the memory and that returns it as `object.Object`

also, think it would be useful for users to declare/clarify the streaming behavior of the `Read`, `ReadChunk`, and `ReadHeader` in the comments